### ISSUE TITLE
ExceptBy doesn't include duplicates by comparer in first set that should not be excluded 

### ISSUE THREAD

**Issue by bsonnino** [dotnet/runtime#64149]
### Description

If the first set has duplicates according to the comparer function, the second one is removed from the resulting set, thus being an error.
The resulting set should include all duplicates that are not in the first set

### Reproduction Steps

If you have something like this:

```
public record Person(string Name, int Age);

var people = new List<Person>
{
    new Person("John", 30 ),
    new Person("Peter", 40),
    new Person("Mary", 20 ),
    new Person("Jane", 30 ),
    new Person("Larry", 50),
    new Person("Anne", 50 ),
    new Person("Paul", 20),
};

var excludedAges = new List<int> {30,40};
var people1 = people.ExceptBy(excludedAges, p => p.Age);
Console.WriteLine($"{string.Join("\n", people1)}");
```

### Expected behavior

```
Person { Name = Mary, Age = 20 }
Person { Name = Larry, Age = 50 }
Person { Name = Anne, Age = 50 }
Person { Name = Paul, Age = 20 }
```

### Actual behavior

``` 
Person { Name = Mary, Age = 20 }
Person { Name = Larry, Age = 50 }
```

### Known Workarounds

var people2 = people.Where(p => !excludedAges.Contains(p.Age));
Console.WriteLine($"{string.Join("\n", people2)}");

### Configuration

.NET 6.0.101
Windows 11 21H2
x64

### Other information

The code in Except.cs (line 94) uses a HashSet, thus eliminating the duplicates due to the comparer, which are not real duplicates

**Comment by Clockwork-Muse** [comment]
`Except` - and by extension, `ExceptBy` - is one of the "[Set Operations](https://en.wikipedia.org/wiki/Set_(mathematics))" (along with `Union` and `Intersect`), and duplicates are always removed (this behavior is standard among pretty much all languages/libraries, because it's such a fundamental concept).

If your use case requires retaining duplicates, you should use `Where` instead.

<hr />

@stephentoub 
That aside, the documentation for `ExceptBy` completely lacks any documentation of this behavior, lacking a `Remarks` section that the other two have, or even the aside about "distinct elements" in the parameter/return section.

**Comment by stephentoub** [comment]
> @stephentoub That aside, the documentation for ExceptBy completely lacks any documentation of this behavior, lacking a Remarks section that the other two have, or even the aside about "distinct elements" in the parameter/return section.

@eiriktsarpalis

**Comment by eiriktsarpalis** [comment]
Yes, a remarks section elaborating a bit more on what "set semantics" means in this case might be warranted (and in particular, how the projector delegate modulates the definition of the source set) might be warranted. Moving to dotnet-api-docs.
